backend/cmm_helpers.ml

```diff
@@ -1287,6 +1287,25 @@ let setfield_unboxed_int64_or_nativeint arr ofs newval dbg =
          [array_indexing log2_size_addr arr ofs dbg; newval],
          dbg ))
 
+(* Getters and setters for unboxed float32 fields *)
+
+let get_field_unboxed_float32 mutability ~block ~index dbg =
```

Comment:
Whether it would work depends on how the backend handles floats, and it appears the upstream big-endian backends don't reorder the bits of floats (there isn't a most/least significant byte anyway). 

(and float32 isn't supported in the upstream backend anyway)

Reply:
Added assertions for consistency with int32 for now

This may matter in the int32 case because loading an int32 field via `Word_int` would require the data to be in the "top" bits and sign extended. But we actually load int32s using `Thirtytwo_signed` which would expect it to be in the closest 4 bytes. And `box_int_gen` is doing the shift, so this is probably already broken on big endian.

Either way doing a load using `Thirtytwo` or `Single` should work regardless of endianness